src/test/java/unit/GcsBucketLifecycle.java

```diff
@@ -0,0 +1,390 @@
+package unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.cli.serialization.userfacing.UFAuthStatus;
+import com.google.api.client.util.DateTime;
+import com.google.cloud.Identity;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.StorageClass;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnit;
+import harness.utils.ExternalGCSBuckets;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for specifying lifecycle rules for controlled GCS buckets. */
+@Tag("unit")
+public class GcsBucketLifecycle extends SingleWorkspaceUnit {
+  // external bucket to use for testing the JSON format against GCS directly
+  private Bucket externalBucket;
+
+  @Override
+  @BeforeAll
+  protected void setupOnce() throws IOException {
+    super.setupOnce();
+    externalBucket = ExternalGCSBuckets.createBucket();
+
+    // `terra auth status --format=json`
+    UFAuthStatus authStatus =
+        TestCommand.runAndParseCommandExpectSuccess(UFAuthStatus.class, "auth", "status");
+
+    // grant the user's proxy group write access to the bucket, so we can test calling `terra gsutil
+    // lifecycle` with the same JSON format used for creating controlled bucket resources with
+    // lifecycle rules
+    ExternalGCSBuckets.grantWriteAccess(externalBucket, Identity.group(authStatus.proxyGroupEmail));
+  }
+
+  @Override
+  @AfterAll
+  protected void cleanupOnce() throws IOException {
+    super.cleanupOnce();
+    ExternalGCSBuckets.getStorageClient().delete(externalBucket.getName());
+    externalBucket = null;
+  }
+
+  @Override
+  @BeforeEach
+  protected void setupEachTime() throws IOException {
+    super.setupEachTime();
+
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
```

Comment:
The `login` just sets up credentials for a test user. There are other tests that either login a user that is not the workspace creator, or that don't set a workspace. I note that just to say these calls are not always bundled. For this test they are always done together though, and putting them in the `@BeforeEach` method was to avoid repetition.